### PR TITLE
made checker kademlia aware

### PR DESCRIPTION
**Pull request by wthorp**
What:
Checker now cares about Kademlia. It waits for it to bootstrap. It checks Kadmelia when the overlay doesn't know about a node.

Why:
Checker never knew about Kademlia before. Anything missing from the overlay didn't get checked correctly.

### CODE REVIEW

File: pkg/datarepair/checker/checker.go [hunk 1]
```diff
@@ -182,7 +186,11 @@ func (checker *Checker) getMissingPieces(ctx context.Context, pieces []*pb.Remot
 	}
 
 	for i, node := range nodes {
-		if node == nil || !checker.overlay.IsOnline(node) || !checker.overlay.IsHealthy(node) {
+		if node == nil {
+			if _, err = checker.kad.FetchPeerIdentity(ctx, nodeIDs[i]); err != nil {
```
Comment (navillasa) [inline pkg/datarepair/checker/checker.go]: why are we fetching peer identity is the node is nil? just curious

**Comment by egonelbre** [comment]
How could this ever happen?

When we have something in the pointerdb, then we also should have it in overlay, otherwise it wouldn't have been selected. Since we must not delete things from overlay nodes table, then it should always be there.

File: pkg/datarepair/checker/checker.go (hunk 1 above)
Comment (wthorp) [reply pkg/datarepair/checker/checker.go]: a nil node means that the overlay doesn't have a record at all.  Its likely perfectly fine, but for whatever reason we don't have a record of it.  I'm still not sure how this is even possible, given that if pointerdb knows of a node, I would think that overlay should know of the node.  
Nevertheless, we are getting nils from overlay.  :/

**Comment by wthorp** [comment]
I think this is totally unnecessary / incoherent

**Pull request closed without merging**